### DISCUSSION TITLE
Easier way to do a bunch of incompatibilities

### DISCUSSION THREAD

**Discussion by jewvenchy** [nftchef/art-engine#269] [Ideas]
i was wondering if there's a way to set incompatibilities but for a whole batch

for example I have sets of color divided into 3 "types" ie base, bright, soft.

inside of these color sets there are 5 diff traits exported in like 20+ colors specific to that color set. so it's:

``` base > trait A > 20 colors
base > trait B > 20 colors
bright > trait A > 20 colors
bright > trait B > 20 colors
soft > trait A > 20 colors
soft > trait B > 20 colors
and so on for 5 traits (A through E)
```

i want it so that traits A-E are never the same color. for example:

token 1 is the base color set, it has trait A as color 1, so trait B-E can never be color 1
token 1 trait B is color 5, so trait C-E can't be color 1 or color 5

is there a better way to do this or anything I can do besides doing it individually for literally every single layer?

ie:

```js
const incompatible = {
Trait A Red: ["Trait B Red", "Trait C Red", "Trait D Red", "Trait E Red"],
"Trait B Red": ["Trait A Red", "Trait C Red", "Trait D Red", "Trait E Red"],
"Trait C Red": ["Trait A Red", "Trait B Red", "Trait D Red", "Trait E Red"],
};
```

thank you very much

**Comment by PhosphoriaArtist** [comment]
There isn't a better way right now. You either have to do exhaustive incompatible lists or duplicating/rearranging multiple nested folder structures. Complicated heuristics or generation 'rules' aren't really a feature for large sets of items/traits.

I've asked someone to look into implementing a "weightmap" type function into the generator for the exact same issues you are having, but that might be a while away yet :)

**Comment by cryptofannatic** [comment]
any update on the implementation @PhosphoriaArtist
